HW1/P6/P6.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+from functools import partial
+import findspark
+findspark.init()
+import pyspark
+
+
+def pre_process_data( filename, sc ):
+    wlist = sc.textFile( filename )
+    wlist_words = wlist.flatMap( lambda x: x.split() )
+    wlist_words_filter = wlist_words.filter( lambda x: not( x.isdigit() or x.isupper() or ( x[:-1].isupper() and x[-1:] == "." ) ) )
+    wlist_words_index = wlist_words_filter.zipWithIndex().map( lambda (x,y): (y,x) )
+    wlist_words_index_2 = wlist_words_index
+    size = wlist_words_index_2.count()
+    wlist_words_index_2 = wlist_words_index_2.map( lambda ( x, y ): ( x-1, y ) if x != 0 else ( size, y ) )
```

Comment:
Line 13 is unnecessary. You could call .count() on the original and use the original as the base for 2